mitodl/micromasters
# Added course team contact email API endpoint

#### What are the relevant tickets?

Closes #2375 

#### What's this PR do?

Adds an API endpoint in the backend to send an email to a course team

#### Where should the reviewer start?

Probably `mail/api.py`

#### How should this be manually tested?

The following snippet can be run (with 'YOUR_USERNAME' and 'SOME_PROGRAM_TITLE' substituted for your preferred username and program title) to send a course contact email. You should run this with `MAILGUN_RECIPIENT_OVERRIDE` set in your .env file so you receive the email.

```python
from courses.models import Course
from django.contrib.auth.models import User
from django.core.urlresolvers import reverse
from django.test.client import Client
#
user = User.objects.get(username='YOUR_USERNAME')
course = Course.objects.filter(program__title='SOME_PROGRAM_TITLE').first()
course.contact_email = 'courseteam@example.com'
course.save()
url = reverse('course_team_mail_api', kwargs={'course_id': course.id})
client = Client()
client.force_login(user)
resp = client.post(
    url,
    data={'email_subject': 'API Email Subject', 'email_body': 'API Email Body'},
)
resp.json()
```
